quipucords/fingerprinter/__init__.py

```diff
@@ -32,203 +52,483 @@ def process_fact_collection(sender, instance, **kwargs):
-
-        if fact.get('etc_release_version'):
-            fingerprint['os_version'] = fact['etc_release_version']
-
-        if fact.get('etc_release_release'):
-            fingerprint['os_release'] = fact['etc_release_release']
-
-        # Set bios UUID
-        if fact.get('dmi_system_uuid'):
-            fingerprint['bios_uuid'] = fact['dmi_system_uuid']
-
-        # Set subscription manager id
-        if fact.get('subman_virt_uuid'):
-            fingerprint['subscription_manager_id'] = fact['subman_virt_uuid']
-
-        # Set connection information
-        if fact.get('connection_uuid'):
-            fingerprint['connection_uuid'] = fact['connection_uuid']
-
-        if fact.get('connection_host'):
-            fingerprint['connection_host'] = fact['connection_host']
-
-        if fact.get('connection_port'):
-            fingerprint['connection_port'] = fact['connection_port']
-
-        # Set CPU facts
-        if fact.get('cpu_count'):
-            fingerprint['cpu_count'] = fact['cpu_count']
-
-        if fact.get('cpu_core_per_socket'):
-            fingerprint['cpu_core_per_socket'] = fact['cpu_core_per_socket']
-
-        if fact.get('cpu_siblings'):
-            fingerprint['cpu_siblings'] = fact['cpu_siblings']
-
-        if fact.get('cpu_hyperthreading') is not None:
-            fingerprint['cpu_hyperthreading'] = fact['cpu_hyperthreading']
-
-        if fact.get('cpu_socket_count'):
-            fingerprint['cpu_socket_count'] = fact['cpu_socket_count']
-
-        if fact.get('cpu_core_count'):
-            fingerprint['cpu_core_count'] = fact['cpu_core_count']
-
-        # Determine system_creation_date
-        system_creation_date = None
-        if fact.get('date_anaconda_log'):
-            system_creation_date = datetime.strptime(
-                fact['date_anaconda_log'], '%Y-%m-%d')
-
-        if system_creation_date and fact.get('date_yum_history'):
-            date_yum_history = datetime.strptime(
-                fact['date_yum_history'], '%Y-%m-%d')
-            if date_yum_history < system_creation_date:
-                system_creation_date = date_yum_history
-
-        if system_creation_date:
-            fingerprint['system_creation_date'] = system_creation_date.date()
-
-        # Determine if running on VM or bare metal
-        if fact.get('virt_what_type') or fact.get('virt_type'):
-            if fact.get('virt_what_type') == 'bare metal':
-                fingerprint['infrastructure_type'] = 'bare_metal'
-            elif fact.get('virt_type'):
-                fingerprint['infrastructure_type'] = 'virtualized'
+    network_fingerprints = []
+    vcenter_fingerprints = []
+    for source in raw_facts['sources']:
+        source_fingerprints = _process_source(
+            raw_facts['fact_collection_id'],
+            source)
+        if source['source_type'] == Source.NETWORK_SOURCE_TYPE:
+            network_fingerprints = network_fingerprints +\
```

Comment:
can't you just do `network_fingerprints += source_fingerprints`?

Reply:
Good point